### PR TITLE
Record inside panache

### PR DESCRIPTION
**Pull request by loicmathieu**
This PR move Panache build steps to use the Computing index to handle `java.lang.Record` this is a recommandation of @Ladicek .

This PR document that records are supported and a good match for projection classes.

This PR adds a Java 17 integration test, it contains 2 tests for using record as projection class for both MongoDB with Panache and Hibernate with Panache, more tests could be added for more Java 17 functionalities later if needed.

### CODE REVIEW

File: integration-tests/pom.xml [hunk 1]
```diff
@@ -340,6 +340,20 @@
...
         </profile>
 
+        <!-- Java 17 test profile that restricts running in 17+ JVM version -->
+        <profile>
+            <id>java-17</id>
+            <activation>
+                <property>
+                    <name>!no-test-modules</name>
+                </property>
+                <jdk>(17,)</jdk>
```
Comment (gsmet) [inline integration-tests/pom.xml:350]: I would have expected:
```suggestion
                <jdk>[17,)</jdk>
```
Why did you use a `(`?

**Review by gsmet — DISMISSED** [review]

**Review by FroMage — APPROVED** [review]
LGTM, once you figure out the Maven syntax that @gsmet raised :)

File: integration-tests/pom.xml (hunk 1 above)
Comment (loicmathieu) [reply integration-tests/pom.xml:350]: @gsmet because ... I made a mistake ;)
Thanks for pointing this out, I'll update it.

**Comment by loicmathieu** [comment]
@gsmet I applied your suggestions and the build it green ! Can we merge it ?

**Pull request merged**